src/sctools/metrics/aggregator.py

```diff
@@ -110,62 +133,63 @@ def finalize(self):
         return sum(1 for base in quality_sequence if base > threshold) / len(quality_sequence)
 
-    # def is_noise(self, record: pysam.AlignedSegment) -> bool:
-    #     """returns True if the record is noise (polymeric or lots of Ns, read 10x code)"""
-    #     raise NotImplementedError
-
-    @staticmethod
-    def is_duplicate(record: pysam.AlignedSegment) -> bool:
-        return record.is_duplicate
+    def is_noise(self, record: pysam.AlignedSegment) -> bool:
+        """returns True if the record is noise (polymeric or lots of Ns, read 10x code)"""
+        return NotImplemented  # todo required because 10x measures this
 
     def parse_molecule(
             self, tags: Sequence[str], records: Iterable[pysam.AlignedSegment]) -> None:
-        """Parse records of a molecule, storing the resulting data
+        """Parse information from all records of a molecule, aggregating the resulting data
 
         :param tags: all the tags that define this molecule
-        :param records: the records associated with the molecule
+        :param records: the sam records associated with the molecule
         """
         for record in records:
 
-            # todo think about how I could use the duplicate tag to simplify this class
+            # todo think about how I could use the duplicate tag to reduce computation; duplicates
+            # should normally come in order in a sorted file
 
-            # call the parse_extra_fields function to extract subclass-specific information
-            # todo might need a callback if this function has knowledge about termination conditions
+            # extract sub-class-specific information
             self.parse_extra_fields(tags=tags, record=record)
 
             self.n_reads += 1
-            # self.noise_reads += self.is_noise(record)
-
-            # get components that define a unique sequence fragment
-            position: int = record.pos
-            strand: bool = record.is_reverse
-            reference: int = record.reference_id
-            self._fragment_histogram[reference, position, strand, tags] += 1
+            # self.noise_reads += self.is_noise(record)  # todo implement me
 
-            self._molecule_histogram[tags] += 1  # might also need gene here (or cell?)
+            # the tags passed to this function define a molecule, this increments the counter,
+            # identifying a new molecule only if a new tag combination is observed
+            self._molecule_histogram[tags] += 1
 
             self._molecule_barcode_fraction_bases_above_30.update(
                 self.quality_above_threshold(
-                    30, self.quality_string_to_numeric(record.get_tag('UY'))
-                )
-            )
+                    30, self.quality_string_to_numeric(record.get_tag('UY'))))
 
             self.perfect_molecule_barcodes += record.get_tag('UR') == record.get_tag('UB')
```

Comment:
Please explain what you mean here. I am intrigued.

Reply:
Somewhere that holds all of these static tag strings like

```
CELL_BARCODE_TAG = 'CB'
CELL_BARCODE_QUAL_TAG = 'CY'
MOLECULAR_BARCODE_TAG = 'UB'
MOLECULAR_BARCODE_QUAL_TAG = 'UY'
...
...
```
and refer to these values instead of the explicit values